setup.py

```diff
@@ -60,6 +60,7 @@ def get_description():
                     'uproot>=3.11.0,<4',
                     'awkward1>=0.4.3,<1',
                     'uproot4>=0.0.17,<1',
+                    'pyarrow>=0.10.0,!=0.14.0',
```

Comment:
pyarrow is up to 2.0.0 now, are we confident this requirement can be as loose as it is?

Reply:
Awkward's requirement on pyarrow is `>=1.0.0` in `awkward1==0.4.3` and `>=2.0.0` in `awkward1==0.4.4`:

https://github.com/scikit-hep/awkward-1.0/commit/d92f75291906b5dc29447a9c6d5b06dbe0b1efa2

Although it's an optional dependency, only loaded at runtime if used, the `ak.to_arrow`/`ak.from_arrow` functions fail if not 2.0 (now). The same PR (above) fixes some errors in UnionArray handling, probably not directly relevant for NanoEvents.

Reply:
I'll push it up to 1.0.0 for coffea.